mmrazor/engine/runner/mixins.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Any
+
+import torch
+import torch.distributed as dist
+from mmengine import autocast
+from mmengine.runner import Runner
+from torch import Tensor
+from torch.nn.modules.batchnorm import _BatchNorm
+from torch.utils.data import DataLoader
+
+
+class AverageMeter:
+    """Computes and stores the average and current value."""
+
+    def __init__(self) -> None:
+        self.reset()
+
+    def reset(self) -> None:
+        self.avg: Tensor = 0
+        self.sum: Tensor = 0
+        self.count: int = 0
+
+    def update(self, val: Any, batch_size: int = 1) -> None:
+        if dist.is_initialized() and dist.is_available():
+            dist.all_reduce(val, dist.ReduceOp.SUM, async_op=False)
+            batch_size_tensor = torch.tensor([batch_size], device=val.device)
+            dist.all_reduce(
+                batch_size_tensor, dist.ReduceOp.SUM, async_op=False)
+            total_batch_size = batch_size_tensor.item()
+
+            val /= (total_batch_size / batch_size)
+            batch_size = total_batch_size
+
+        self.sum += val * batch_size
+        self.count += batch_size
+        self.avg = self.sum / self.count
+
+
+class CalibrateBNMixin:
+    runner: Runner
+    fp16: bool
+
+    @torch.no_grad()
+    def calibrate_bn_statistics(self,
```

Comment:
In some cases (`train`->`val`->`train`), we may want to resume the statics after `val` mode.
A context manager could be used to handle this.

Reply:
Will be fixed in another pr.